salt/modules/boto_cloudfront.py

```diff
@@ -0,0 +1,441 @@
+    The caller is responsible for all boto-related error handling.
+
+    name
+        (Optional) Only yield the distribution with the given name
+    '''
+    for dl in conn.get_paginator('list_distributions').paginate():
+        if 'Items' not in dl['DistributionList']:
+            # If there are no items, AWS omits the `Items` key for some reason
+            continue
+        for partial_dist in dl['DistributionList']['Items']:
+            tags = conn.list_tags_for_resource(Resource=partial_dist['ARN'])
+            tags = dict((kv['Key'], kv['Value']) for kv in tags['Tags']['Items'])
+
+            id_ = partial_dist['Id']
+            if 'Name' not in tags:
+                log.warning(
+                    'CloudFront distribution {0} has no Name tag.'.format(id_),
+                )
+                continue
+            d_name = tags.pop('Name', None)
+            if name is not None and d_name != name:
+                continue
+
+            # NOTE: list_distributions() returns a DistributionList,
+            # which nominally contains a list of Distribution objects.
+            # However, they are mangled in that they are missing values
+            # (`Logging`, `ActiveTrustedSigners`, and `ETag` keys)
+            # and moreover flatten the normally nested DistributionConfig
+            # attributes to the top level.
+            # Hence, we must call get_distribution() to get the full object,
+            # and we cache these objects to help lessen API calls.
+            distribution = _cache_id(
+                'cloudfront',
+                sub_resource=d_name,
+                region=region,
+                key=key,
+                keyid=keyid,
+                profile=profile,
+            )
+            if distribution:
+                yield (d_name, distribution)
+                continue
+
+            dist_with_etag = conn.get_distribution(Id=id_)
+            distribution = {
+                'distribution': dist_with_etag['Distribution'],
+                'etag': dist_with_etag['ETag'],
+                'tags': tags,
+            }
+            _cache_id(
+                'cloudfront',
+                sub_resource=d_name,
+                resource_id=distribution,
+                region=region,
+                key=key,
+                keyid=keyid,
+                profile=profile,
+            )
+            yield (d_name, distribution)
+
+
+def get_distribution(name, region=None, key=None, keyid=None, profile=None):
+    '''
+    Get information about a CloudFront distribution (configuration, tags) with a given name.
+
+    name
+        Name of the CloudFront distribution
+
+    region
+        Region to connect to
+
+    key
+        Secret key to use
+
+    keyid
+        Access key to use
+
+    profile
+        A dict with region, key, and keyid,
+        or a pillar key (string) that contains such a dict.
+
+    CLI Example::
```

Comment:
This is super minor, but the double `::` here will cause problems for the docs to look pretty. Just one `:` should be good.